### PR TITLE
[Core][aDAG] Set buffer size to 1 for regression

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

There's a regression with buffer size 10. I am going to investigate but I will revert it to buffer size 1 for now until further investigation. 
With buffer size 1, regression seems to be gone https://buildkite.com/ray-project/release/builds/22594#0191ed4b-5477-45ff-be9e-6e098b5fbb3c. probably some sort of contention or sth like that

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/dag/compiled_dag_node.py [hunk 1]
```diff
@@ -661,7 +661,7 @@ def __init__(
             # We conservatively set num_shm_buffers to _max_inflight_executions.
             # It means that the DAG can be underutilized, but it guarantees there's
             # no false positive timeouts.
-            num_shm_buffers=self._max_inflight_executions,
+            num_shm_buffers=1,
```
Comment (ruisearch42) [inline python/ray/dag/compiled_dag_node.py:664]: Can you update the comment above and add a TODO?

**Review by ruisearch42 — APPROVED** [review]

File: python/ray/dag/compiled_dag_node.py (hunk 1 above)
Comment (rkooo567) [reply python/ray/dag/compiled_dag_node.py:664]: I will fix this this week, so it may be okay?

Comment (ruisearch42) [reply python/ray/dag/compiled_dag_node.py:664]: Yes, feel free to merge first if urgent.

**Review by kevin85421 — APPROVED** [review]
Do we have an issue to track the regression caused by the buffer PR? If not, would you mind opening an issue to track the progress?

**Comment by rkooo567** [comment]
It is in my sprint task.

**Pull request merged**